crates/bevy_ecs/src/world/error.rs

```diff
@@ -25,60 +23,12 @@ pub enum EntityComponentError {
 }
 
 /// An error that occurs when fetching entities mutably from a world.
-#[derive(Clone, Copy)]
-pub enum EntityFetchError<'w> {
+#[derive(Error, Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EntityFetchError {
```

Comment:
Wow I much prefer removing the lifetime + UnsafeWorldCell from this. Can we split this out?